app/src/containers/AppContainer.tsx

```diff
@@ -0,0 +1,27 @@
+import React from "react";
+import App from "../../App";
+import { useLayoutEffect } from "react";
+import { useTheme } from "../ui/hooks/useTheme";
+import { Theme } from "../ui/hooks/useTheme";
+import styles from "./AppContainer.module.css";
+import CardContainer from "./CardContainer";
+
+const AppContainer = () => {
+  const { restoreTheme, setTheme, getTheme } = useTheme();
+  const handleCheckboxChange = () => {
+    setTheme(getTheme() === Theme.Dark ? Theme.Light : Theme.Dark);
+  };
+  const isCurrentThemeLight = getTheme() === "light" ? true : false;
+
+  useLayoutEffect(() => {
```

Comment:
lets use here `useEffect` cause it requires less computations comaring to `useLayoutEffect`